src/actions.cpp

```diff
@@ -479,8 +479,14 @@ void evm_contract::process_tx(const runtime_config& rc, eosio::name miner, const
 
     silkworm::ExecutionProcessor ep{block, engine, state, *found_chain_config->second, gas_params};
 
-    check(tx.max_priority_fee_per_gas == tx.max_fee_per_gas, "max_priority_fee_per_gas must be equal to max_fee_per_gas");
-    check(tx.max_fee_per_gas >= _config->get_gas_price(), "gas price is too low");
+    if (current_version >= 1) {
+        auto inclusion_price = std::min(tx.max_priority_fee_per_gas, tx.max_fee_per_gas - *base_fee_per_gas);
+        eosio::check(inclusion_price >= (min_inclusion_price.has_value() ? *min_inclusion_price : 0), "inclusion price too low");
```

Comment:
We can also express more explicitly that we only validate in case `min_inclusion_price` is defined
```
eosio::check(!min_inclusion_price.has_value() || inclusion_price >= min_inclusion_price.value(), "inclusion price too low");
```